spaceR/spaceR/Controllers/RocketsControlller.swift

```diff
@@ -11,30 +11,28 @@ import UIKit
 
 class RocketsController: UIViewController {
     
-    @IBOutlet weak var tableView: UITableView!
-    
-    @IBOutlet weak var emailIdLabel: UILabel!
-    var emailText = ""
-    private var rockets: [Rocket] = []
+    @IBOutlet weak var rocketTableView: UITableView!
+    private lazy var rocketViewModel = RocketViewModel()
     
     override func viewDidLoad() {
-        super.viewDidLoad()
-        self.emailIdLabel.text = self.emailText
-        getAllRockets()
-        setupTableView()
+        getAllRocketsFromAPI()
+        setTableView()
     }
     
-    private func setupTableView() {
-        self.tableView.delegate = self
-        self.tableView.dataSource = self
+    
+    func setTableView() {
+        rocketTableView.delegate = self
+        rocketTableView.dataSource = self
     }
     
-    func getAllRockets() {
+    func getAllRocketsFromAPI() {
         URLSession.shared.getAllRocketsEndpointURL(url: Constants.getAllRocketsUrl, model: [Rocket].self){ [weak self]result in
             switch result {
-            case .success(let userArray):
-                self?.rockets = userArray
-                self?.tableView.reloadData()
+            case .success(let rocketsArray):
+                self?.rocketViewModel.setAllRockets(rockets: rocketsArray)
+                DispatchQueue.main.async {
+                    self?.rocketTableView.reloadData()
+                }
             case .failure(let error):
                 print(error)
```

Comment:
This we can maybe handle this a little bit better 😄 maybe just an alert to the user to let them know that something has went wrong.

Reply:
Okay, will add some error handeling